osu.Game/Screens/Play/PlayerSettings/BeatmapOffsetControl.cs

```diff
@@ -0,0 +1,212 @@
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Threading.Tasks;
+using osu.Framework.Allocation;
+using osu.Framework.Bindables;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Utils;
+using osu.Game.Beatmaps;
+using osu.Game.Database;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Containers;
+using osu.Game.Graphics.Sprites;
+using osu.Game.Overlays.Settings;
+using osu.Game.Rulesets.Scoring;
+using osu.Game.Scoring;
+using osu.Game.Screens.Ranking.Statistics;
+using osuTK;
+using osu.Game.Localisation;
+using osu.Game.Rulesets.Mods;
+
+namespace osu.Game.Screens.Play.PlayerSettings
+{
+    public class BeatmapOffsetControl : CompositeDrawable
+    {
+        public Bindable<ScoreInfo> ReferenceScore { get; } = new Bindable<ScoreInfo>();
+
+        public BindableDouble Current { get; } = new BindableDouble
+        {
+            Default = 0,
+            Value = 0,
+            MinValue = -50,
+            MaxValue = 50,
+            Precision = 0.1,
```

Comment:
Curious about the usage of bounds and precision here - is it intentional that all these quantities are one tenth of what global offset has?

Reply:
Not intentional, just set what I felt was a valid range for adjustments for beatmaps. Going beyond 50ms would suggest much more serious issues than should be fixed with an offset control. Will check what the limits are on stable and circle back.